lib/services/services-db.js

```diff
@@ -0,0 +1,26 @@
+"use strict";
+
+var ServiceDBSchema = require("../models/services-schema");
+var Genesis         = require("genesis");
+var uuid            = require("uuid");
+var Bluebird        = require("bluebird");
+
+Bluebird.promisifyAll(uuid);
+
+function Instance (mapper) {
+	if (!mapper) {
+		mapper = new Genesis.MemoryMapper();
+	}
+
+	this.createInstance = function (ami, type) {
+		if (arguments.length > 2) {
```

Comment:
Although it's good to be careful, nothing bad would happen if you pass too many parameters to a JavaScript function. I wouldn't say this is necessary to guard against.